gorm-tools-rest/src/main/groovy/gorm/tools/rest/controller/RestRepositoryApi.groovy

```diff
@@ -230,36 +236,24 @@ trait RestRepositoryApi<D> implements RestApiController {
     List getqSearchIncludes() { [] }
 
     void handleException(RuntimeException e) {
-        //log.error e.message
         if( e instanceof EntityNotFoundException){
-            callRender(status: NOT_FOUND, e.message)
+            respond([view: '/errors/_errors'], new ApiError(status:NOT_FOUND, title: "Not Found", detail:e.message))
         }
         else if( e instanceof EntityValidationException ){
-            String defaultMessage = e.messageMap.defaultMessage as String
-            // log.info m
-            respond([view: '/errors/_errors'], [errors: e.errors, message: defaultMessage, renderArgs: [:]])
-            //callRender(status: UNPROCESSABLE_ENTITY, m)
+            respond([view: '/errors/_errors422'], new ApiValidationError(UNPROCESSABLE_ENTITY, "Validation Error", e.defaultMessage, e.errors))
         }
         else if( e instanceof ValidationException ){
-            String defaultMessage = e.message
-            respond([view: '/errors/_errors'], [errors: e.errors, message: defaultMessage, renderArgs: [:]])
+            respond([view: '/errors/_errors422'], new ApiValidationError(UNPROCESSABLE_ENTITY, "Validation Error", e.message, e.errors))
+        }
+        else if(e instanceof DataAccessException ){
+            respond([view: '/errors/_errors'], new ApiError(status:UNPROCESSABLE_ENTITY, title: "Data Access Exception", detail:e.message))
         }
-        else if( e instanceof OptimisticLockingFailureException ){
```

Comment:
why did we remove it ?

Reply:
Because it is caught under DataAccessException, and as we did not set different status or message for it.... the case is handled under catch DataAcessException